<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [a1, aliengo, go1, laikago]"/>

    <param name="legged_robot_description" command="$(find xacro)/xacro $(find legged_unitree_description)/urdf/robot.xacro
       robot_type:=$(arg robot_type)
    "/>
    <node name="generate_urdf" pkg="legged_common" type="generate_urdf.sh" output="screen"
          args="$(find legged_unitree_description)/urdf/robot.xacro $(arg robot_type)"/>

    <rosparam file="$(find legged_ros_hw)/config/default.yaml" command="load"/>

    <param name="robot_type" value="$(arg robot_type)"/>
    <node name="legged_ros_hw" pkg="legged_ros_hw" type="legged_ros_hw"
          output="screen"
          respawn="false"
          clear_params="true"/>
</launch>

